ctapipe/io/datawriter.py

```diff
@@ -647,6 +647,11 @@ def _write_dl1_telescope_events(
         """
 
         # write the telescope tables
+        # trigger info
```

Comment:
is there any functional change in moving this block before the pointing loop?

Reply:
This PR basically has to independent fixes for the issue #2356, each fix it on its own.

Writing the trigger data first prevents the possible mutation in the loop below for the pointing to affect the file written and the change in the software trigger now removes all telescope-event related data for telescopes that are removed.

Strictly speaking this change wouldn't be necessary anymore, but I also found the order more logical.